src/algebra/group_power/lemmas.lean

```diff
@@ -289,6 +297,17 @@ begin
   exact lt_irrefl _ (lt_of_le_of_lt (nsmul_le_nsmul (le_of_lt ha) $ not_lt.mp H) h)
 end
 
+lemma gsmul_eq_gsmul_iff {a b : A} {m : ℤ} (hm : 0 < m) : m • a = m • b ↔ a = b :=
```

Comment:
Isn't `m ≠ 0` sufficient here? I think this should be called something like `gsmul_right_inj` to be consistent with `mul_right_inj`

Reply:
I think this name is worth adding since it matches `gsmul_lt_gsmul_iff` and `gmsul_le_gsmul_iff` in the same file. I've added `gsmul_left_injective` though as well, and could add a `gsmul_left_inj` if you like as an alias.

Having written this, I've realized I might have `left` and `right` backwards...

Reply:
I think the naming convention is that these `iff` lemmas about injectivity should be called something ending with `_inj`. I vaguely remember a zulip discussion about this, but I can't find it.

Reply:
You're right, I've added that spelling too.

I just checked; "left" is consistent with the `smul_left_injective` lemma we have elsewhere.

Reply:
It should be `right` according to the naming convention. https://leanprover-community.github.io/contribute/naming.html (Search for `right_inj`). It surprised me as well. You should probably change `smul_left_injective` at the same time.

Reply:
I've gone ahead and fixed the names in #8659 and this PR.